cli/types/cypress.d.ts

```diff
@@ -461,8 +461,9 @@ declare namespace Cypress {
     Commands: {
       add<T extends keyof Chainable>(name: T, fn: CommandFn<T>): void
       add<T extends keyof Chainable>(name: T, options: CommandOptions & {prevSubject: false}, fn: CommandFn<T>): void
+      add<T extends keyof Chainable>(name: T, options: CommandOptions & {prevSubject: true}, fn: CommandFnWithSubject<T, PrevSubjectMap['element']>): void
```

Comment:
We can't be sure that an element will be the previous subject here. It would depend on what command this is chained off of for it to be an element or not.

For example:
- `cy.readFile` yields the file contents
- `cy.tick` yields a clock object
- `cy.wrap` yields the object it was called with which could be an anything like an element or string
- `cy.log` and `cy.writeFile` yields null

If a new command is being added and is expecting the subject to be an element, is there a reason the types can't specific the `prevSubject: 'element'` instead of `prevSubject: true`?

Reply:
Yes, we can specify the `prevSubject: 'element'` instead. 
So whenever `prevSubject: true` it means that the command could be chainable.
 
I think it is better to be generic; so the user could define his type: 

```sugestion 
      add<T extends keyof Chainable, S = unknown>(name: T, options: CommandOptions & {prevSubject: true}, fn: CommandFnWithSubject<T, S>): void
```